addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/DirectoryBrowsingScanRule.java

```diff
@@ -105,21 +105,29 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
         if (evidence != null && evidence.length() > 0) {
             // we found something
-            newAlert()
-                    .setName(getName() + " - " + server)
-                    .setRisk(Alert.RISK_MEDIUM)
-                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                    .setDescription(getDescription() + " - " + server)
-                    .setOtherInfo(getExtraInfo(msg, evidence))
-                    .setSolution(getSolution())
-                    .setReference(getReference())
-                    .setEvidence(evidence)
-                    .setCweId(548) // Information Exposure Through Directory Listing
-                    .setWascId(16) // Directory Indexing
-                    .raise();
+            this.buildAlert(server, evidence, msg).raise();
         }
     }
 
+    private AlertBuilder buildAlert(String server, String evidence, HttpMessage msg) {
+        return newAlert()
+                .setName(getName() + " - " + server)
+                .setRisk(Alert.RISK_MEDIUM)
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setDescription(getDescription() + " - " + server)
+                .setOtherInfo(getExtraInfo(msg, evidence))
+                .setSolution(getSolution())
+                .setReference(getReference())
+                .setEvidence(evidence)
+                .setCweId(548) // Information Exposure Through Directory Listing
+                .setWascId(16); // Directory Indexing
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(buildAlert("Apache 2", "Microsoft IIS", new HttpMessage()).build());
```

Comment:
It wouldn't make sense for the values to be mixed. `server` will be the value out of the hashmap while evidence will be the actual string that the regex pattern matched against in the traffic.

You could temporarily print the values from the alert generated in one of the unit tests as a quick way to see. (They're likely both the same or one a substring of the other).

Reply:
The message is not actually used, good time to just remove it.

Reply:
Agreed

Reply:
In the scan rule and existing unittests.